src/index.js

```diff
@@ -19,35 +19,35 @@ export default function loader(content) {
   });
 
   let outputPath = '';
+  const filePath = this.resourcePath;
 
   if (options.outputPath) {
     // support functions as outputPath to generate them dynamically
     outputPath = (
-      typeof options.outputPath === 'function' ? options.outputPath(url) : options.outputPath
+      typeof options.outputPath === 'function' ? options.outputPath(url, filePath) : options.outputPath
```

Comment:
Please give an example for `url` && `filePath`

Reply:
This is my config in webpack:
```
{
      test: /\.(png|jpg|gif|svg|eot|ttf|woff|woff2)$/,
      loader: 'url-loader',
      options: {
        limit: 10000,
        outputPath: (rPath, fullPath) => {
          return path.relative(__dirname, fullPath).replace(/\\/g, '/');
        },
        useRelativePath: true,
        name: fullPath => {
          if(fullPath.indexOf('builder') > -1){
            return path.relative(path.join(__dirname, 'builder'), fullPath).replace(/\\/g, '/');
          }else{
            return path.relative(path.join(__dirname, 'stemapp'), fullPath).replace(/\\/g, '/');
          }
        }
      }
    }
```
This is the CSS code:
```
.bg-img{
    background-image: url('webapp.png')
  }
```
In this case, the url is `./webapp.png`, the `filePath` is `d:/root_path/..../webapp.png`

Reply:
Please post your file tree, this looks confusing and I don't really get what you're up to tbh 🙃

Reply:
My file tree and the case is a little complex, it's like this:
```
builder
  |_some folders
stemapp
  |_widgets
    |_style.css
    |_webapp.png
```

What I want to do is: let the outputPath return a path but the name return a different path.